lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -1380,23 +1381,43 @@ status_code_e s1ap_mme_generate_ue_context_modification(
   if (ue_ref_p == NULL) {
     OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
   }
+  pdu.present = S1ap_S1AP_PDU_PR_initiatingMessage;
+  pdu.choice.initiatingMessage.procedureCode =
+      S1ap_ProcedureCode_id_UEContextModification;
+  pdu.choice.initiatingMessage.criticality = S1ap_Criticality_reject;
+  pdu.choice.initiatingMessage.value.present =
+      S1ap_InitiatingMessage__value_PR_UEContextModificationRequest;
   container =
       &pdu.choice.initiatingMessage.value.choice.UEContextModificationRequest;
-  pdu.choice.successfulOutcome.procedureCode =
-      S1ap_ProcedureCode_id_UEContextModification;
-  pdu.present = S1ap_S1AP_PDU_PR_initiatingMessage;
+
   /*
    * Fill in ID pair
    */
+  ie = (S1ap_UEContextModificationRequestIEs_t*) calloc(
+      1, sizeof(S1ap_UEContextModificationRequestIEs_t));
+  ie->id          = S1ap_ProtocolIE_ID_id_UE_S1AP_IDs;
+  ie->criticality = S1ap_Criticality_reject;
+  ie->value.present =
+      S1ap_UEContextModificationRequestIEs__value_PR_MME_UE_S1AP_ID;
   ie->value.choice.MME_UE_S1AP_ID = ue_ref_p->mme_ue_s1ap_id;
-  ie->value.choice.ENB_UE_S1AP_ID = ue_ref_p->enb_ue_s1ap_id;
+  ASN_SEQUENCE_ADD(&container->protocolIEs.list, ie);
 
-  S1AP_FIND_PROTOCOLIE_BY_ID(
-      S1ap_UEContextModificationRequestIEs_t, ie, container,
-      S1ap_ProtocolIE_ID_id_RegisteredLAI, true);
+  ie = (S1ap_UEContextModificationRequestIEs_t*) calloc(
+      1, sizeof(S1ap_UEContextModificationRequestIEs_t));
+  ie->id          = S1ap_ProtocolIE_ID_id_UE_S1AP_IDs;
+  ie->criticality = S1ap_Criticality_reject;
+  ie->value.present =
+      S1ap_UEContextModificationRequestIEs__value_PR_ENB_UE_S1AP_ID;
+  ie->value.choice.ENB_UE_S1AP_ID = ue_ref_p->enb_ue_s1ap_id;
+  ASN_SEQUENCE_ADD(&container->protocolIEs.list, ie);
 
   if ((ue_context_mod_req_pP->presencemask & S1AP_UE_CONTEXT_MOD_LAI_PRESENT) ==
       S1AP_UE_CONTEXT_MOD_LAI_PRESENT) {
+    ie = (S1ap_UEContextModificationRequestIEs_t*) calloc(
```

Comment:
how did you verify these changes?

Reply:
At the moment only by unit test, previously calling this function results in a segfault, not sure if we have an integ test or similar that calls this functions, as this is called by handling `S1AP_UE_CONTEXT_MODIFICATION_REQUEST` sent by MME